src/librepgp/stream-write.cpp

```diff
@@ -531,24 +564,77 @@ encrypted_add_recipient(pgp_write_handler_t *handler,
         return RNP_ERROR_NO_SUITABLE_KEY;
     }
 
+#if defined(ENABLE_CRYPTO_REFRESH)
+    /* Crypto Refresh: For X25519/X448 PKESKv3, AES is mandated */
+    if (userkey->alg() == PGP_PKA_X25519 && pkesk_version == PGP_PKSK_V3) {
+        switch (param->ctx->ealg) {
+        case PGP_SA_AES_128:
+        case PGP_SA_AES_192:
+        case PGP_SA_AES_256:
+            break;
+        default:
+            RNP_LOG("attempting to use X25519 and v3 PKESK in combination with a symmetric "
+                    "algorithm that is not AES.");
+            return RNP_ERROR_DECRYPT_FAILED;
+        }
+    }
+#endif
+
     /* Fill pkey */
-    pkey.version = PGP_PKSK_V3;
+    pkey.version = pkesk_version;
     pkey.alg = userkey->alg();
+    /* set key_id (used for PKESK v3) and fingerprint (used for PKESK v6) */
     pkey.key_id = userkey->keyid();
+#if defined(ENABLE_CRYPTO_REFRESH)
+    pkey.fp = userkey->fp();
```

Comment:
As this code line is required only for v6 it would be cleaner to execute it only for that case.